fastdata/core.py

```diff
@@ -44,23 +102,123 @@ def generate(self,
+                futures = [
+                        executor.submit(
+                            self._process_input,
+                            prompt_template,
+                            schema,
+                            temp,
+                            sp,
+                            input_data,
+                        )
+                        for input_data in inputs
+                    ]
+
                 for completed_future in concurrent.futures.as_completed(futures):
                     pbar.update(1)
         return [f.result() for f in futures]
+
+    def generate_to_hf(
+        self,
+        prompt_template: str,
+        inputs: list[dict],
+        schema,
+        repo_id: str,
+        temp: float = 1.0,
+        sp: str = "You are a helpful assistant.",
+        max_workers: int = 64,
+        save_interval: int = 100,
+        commit_every: Union[int, float] = 5,
+        private: bool = False,
+        token: Optional[str] = None,
+        delete_files_after: bool = True,
+    ) -> tuple[str, list[dict]]:
+        """
+        Generate data based on a prompt template and schema, and save it to Hugging Face dataset repository.
+
+        Args:
+            prompt_template (str): The template for generating prompts.
+            inputs (list[dict]): A list of input dictionaries to be processed.
+            schema: The schema to parse the generated data.
+            repo_id (str): The HuggingFace dataset name.
+            temp (float, optional): The temperature for generation. Defaults to 1.0.
+            sp (str, optional): The system prompt for the assistant. Defaults to "You are a helpful assistant.".
+            max_workers (int, optional): The maximum number of worker threads. Defaults to 64.
+            save_interval (int, optional): The batch size at which to save the results. Defaults to 100.
+            commit_every (Union[int, float], optional): The number of minutes between each commit. Defaults to 5.
+            private (bool, optional): Whether the repository is private. Defaults to False.
+            token (Optional[str], optional): The token to use to commit to the repo. Defaults to the token saved on the machine.
+            delete_files_after (bool, optional): Whether to delete files after processing. Defaults to True.
+
+        Returns:
+            tuple[str, list[dict]]: A tuple with the generated repo_id and the list of generated data dictionaries.
+        """
+        dataset_dir = Path(repo_id.replace("/", "_"))
+        dataset_dir.mkdir(parents=True, exist_ok=True)
+        data_dir = dataset_dir / "data"
+        data_dir.mkdir(exist_ok=True)
+
+        try:
+            scheduler = CommitScheduler(
+                repo_id=repo_id,
+                repo_type="dataset",
+                folder_path=dataset_dir,
+                every=commit_every,
+                private=private,
+                token=token,
+            )
+            DatasetCard(
```

Comment:
(nit) maybe add a check to not overwrite the dataset card if it already exists on the Hub.